static/third/bootstrap-typeahead/typeahead.js

```diff
@@ -96,6 +110,34 @@
       return e.keyCode;
   }
 
+  function get_string_diff(string1, string2) {
+    // This function specifies the diff between 2 strings. For example, the
+    // diff between "#ann is for updates" and "#**announce** is for updates"
+    // is "ann" and "**announce**" and the index of the start of the diff is 1
+    const length_diff = string2.length - string1.length;
+
+    let forward_same_till = -1;
+    for(let i = 0; i < Math.min(string1.length, string2.length); i++) {
+      if (string1.charAt(i) === string2.charAt(i)) {
+        forward_same_till = i
+      } else {
+        break;
+      }
+    }
+
+    let backward_same_till = string1.length;
+    for(let i = string1.length - 1; i >= 0; i--) {
+      if (string1.charAt(i) === string2.charAt(i + length_diff)) {
+        backward_same_till = i
+      } else {
+        break;
+      }
+    }
+    const diff1 = string1.substring(forward_same_till+1, backward_same_till);
+    const diff2 = string2.substring(forward_same_till+1, backward_same_till+length_diff)
+    return [diff1, diff2, forward_same_till+1];
```

Comment:
This sort of pure string manipulation function seems like a great function to put in `static/js/util.js` and add a simple node test suite for.

Reply:
Moved it to `util.ts` and added node tests!